apps/wallet-connect/src/hooks/useWalletConnect.tsx

```diff
@@ -143,6 +144,15 @@ const useWalletConnect = () => {
     }
   }, [connector, wcConnect]);
 
+  useEffect(() => {
+    if (connector) {
+      const { accounts } = (connector as Connector) || [];
+      if (accounts.length && !accounts.includes(safe.safeAddress)) {
+        wcDisconnect();
+      }
+    }
+  }, [connector, safe.safeAddress, wcDisconnect]);
+
```

Comment:
WalletConnect has an undocumented `storageId` option which is essentially the key to store the session at:
https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/helpers/types/index.d.ts#L243
https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/clients/core/src/storage.ts#L7

How about using this instead? I think it would be more convenient for those who manage multiple Safes.

Reply:
mmm, and how do you want to use this? This is the id for the current session isn't it? Then when changing safes should be the same i guess. Could you elaborate on this a little more 🙃

Reply:
You can include the wallet address to the key the session is stored

Reply:
so you mean storing multiple sessions and try to get one if the safe address match?

Reply:
exactly

Reply:
Changed the approach and now seems to manage several safes nicely. Thanks 😁